ovs/log/healthcheck_logHandler.py

```diff
@@ -0,0 +1,240 @@
+        """ Init method """
+        pass
+
+    OKBLUE = '\033[94m'
+    OKGREEN = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    SKIP = '\033[95m'
+    ENDC = '\033[0m'
+
+
+class HCLogHandler:
+    """
+    Open vStorage Log Handler
+    """
+
+    def __init__(self, unattended_mode, silent_mode=False):
+        """
+        Init method for the HealthCheck Log handler
+
+        @param unattended_mode: determines the attended modus you are running
+            * unattended run (for monitoring)
+            * attended run (for user)
+        @param silent_mode: determines if you are running in silent mode
+            * silent run (to use in-code)
+
+        @type unattended_mode: bool
+        @type silent_mode: bool
+        """
+
+        # module specific
+        self.module = "utils"
+
+        # load config file
+        with open(Utils().settings_loc) as settings_file:
+            self.settings = json.load(settings_file)
+
+        # fetch from config file
+        self.HEALTHCHECK_DIR = self.settings["healthcheck"]["logging"]["directory"]
+        self.HEALTHCHECK_FILE = self.settings["healthcheck"]["logging"]["file"]
+        self.debug = self.settings["healthcheck"]["debug_mode"]
+
+        # utils log settings (determine modus)
+        if silent_mode:
+            # if silent_mode is true, the unattended is also true
+            self.unattended_mode = True
+            self.silent_mode = True
+        else:
+            self.unattended_mode = unattended_mode
+            self.silent_mode = False
+
+        # HC counters
+        self.failure = 0
+        self.success = 0
+        self.warning = 0
+        self.info = 0
+        self.exception = 0
+        self.skip = 0
+        self.debug = 0
+
+        # result of healthcheck in dict form
+        self.healthcheck_dict = {}
+
+        # create if dir does not exists
+        if not os.path.isdir(self.HEALTHCHECK_DIR):
+            os.makedirs(self.HEALTHCHECK_DIR)
+
+    def logger(self, message, module, log_type, unattended_mode_name, unattended_print_mode=True):
+        """
+        Logs the healthcheck output in a certain way chosen by the user
+
+        @param message: Log message for attended run
+        @param module: describes the module you are logging from
+        @param log_type: describes the log level
+            * failure = 0
+            * success = 1
+            * warning = 2
+            * info = 3
+            * exception = 4
+            * skip = 5
+            * debug = 6
+        @param unattended_mode_name: name for monitoring output
+        @param unattended_print_mode: describes if you want to print the output during a unattended run
+
+        @type message: str
+        @type module: str
+        @type log_type: int
+        @type unattended_print_mode: bool
+        @type unattended_mode_name: str
+        """
+
+        try:
+            target = open('{0}/{1}'.format(self.HEALTHCHECK_DIR, self.HEALTHCHECK_FILE), 'a')
```

Comment:
Use a ```with open(...) as target:```, so you don't need to manually close the file at the end, and to make sure it's also closed when there's an exception.